subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerExecutionHistoryRepository.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.internal.cache.StringInterner;
+import org.gradle.cache.CacheBuilder;
+import org.gradle.cache.CacheRepository;
+import org.gradle.cache.CleanupAction;
+import org.gradle.cache.FileLockManager;
+import org.gradle.cache.PersistentCache;
+import org.gradle.cache.PersistentIndexedCache;
+import org.gradle.cache.PersistentIndexedCacheParameters;
+import org.gradle.cache.internal.CompositeCleanupAction;
+import org.gradle.cache.internal.InMemoryCacheDecoratorFactory;
+import org.gradle.cache.internal.LeastRecentlyUsedCacheCleanup;
+import org.gradle.cache.internal.SingleDepthFilesFinder;
+import org.gradle.internal.concurrent.Stoppable;
+import org.gradle.internal.execution.history.impl.FileCollectionFingerprintSerializer;
+import org.gradle.internal.execution.history.impl.SerializableFileCollectionFingerprint;
+import org.gradle.internal.fingerprint.FileCollectionFingerprint;
+import org.gradle.internal.hash.HashCode;
+import org.gradle.internal.resource.local.FileAccessTimeJournal;
+import org.gradle.internal.resource.local.SingleDepthFileAccessTracker;
+import org.gradle.internal.serialize.BaseSerializerFactory;
+import org.gradle.internal.serialize.Decoder;
+import org.gradle.internal.serialize.Encoder;
+import org.gradle.internal.serialize.HashCodeSerializer;
+import org.gradle.internal.serialize.ListSerializer;
+import org.gradle.internal.serialize.Serializer;
+
+import java.io.File;
+import java.util.Optional;
+
+import static org.gradle.api.internal.artifacts.ivyservice.CacheLayout.TRANSFORMS_META_DATA;
+import static org.gradle.api.internal.artifacts.ivyservice.CacheLayout.TRANSFORMS_STORE;
+import static org.gradle.cache.internal.LeastRecentlyUsedCacheCleanup.DEFAULT_MAX_AGE_IN_DAYS_FOR_RECREATABLE_CACHE_ENTRIES;
+import static org.gradle.cache.internal.filelock.LockOptionsBuilder.mode;
+
+public class DefaultTransformerExecutionHistoryRepository implements TransformerExecutionHistoryRepository, Stoppable {
```

Comment:
I used the same type as we had on `DefaultCachingTransformerExecutor`. I guess `Closeable` is recognized by our service loader infrastructure as well, right? Then I can change it to the built-in type.

Reply:
Looks like our infrastructure supports `Closeable` but not `AutoCloseable`: https://github.com/gradle/gradle/blob/506aa2182392c562c2a0a61e4ad44ccbf8888ff0/subprojects/base-services/src/main/java/org/gradle/internal/concurrent/CompositeStoppable.java#L74-L95

IIUC, you would prefer `Closeable` since the underlying cache is also `Closeable`?

Reply:
https://github.com/gradle/build-cache/issues/1323